[Streams] Agentic onboarding

dgieselaar opened the pull request.

dgieselaar committed:
Promptszzzz

dgieselaar committed:
Promptszzzz

dgieselaar committed:
[Streams] Agent (WIP)

dgieselaar committed:
[Streams] Agent (WIP)

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

dgieselaar committed:
Extract templates programmatically

dgieselaar committed:
Extract templates programmatically

nikitaindik committed:
[Security Solution] Update prebuilt rule upgrade and import test plans to reflect licensing changes (#216003)

**Partially addresses: https://github.com/elastic/kibana/issues/202068**
**Related PR with licensing checks implementation:
https://github.com/elastic/kibana/pull/206079**

## Summary
This PR updates the Prebuilt Rule Upgrade and import/export test plans
to reflect [recent
changes](https://github.com/elastic/kibana/pull/206079) related to
licensing.

---------

Co-authored-by: Maxim Palenov <maxim.palenov@elastic.co>

mistic committed:
chore(NA): update versions after v8.18.3 bump (#221961)

This PR is a simple update of our versions file after the recent bumps.

dej611 committed:
[Lens] Keep suggestions up to date with the main workspace (#221901)

## Summary

Fixes #178571 

This PR reverts #221248 adding the missing part on it to make the bug
found in #178571 disappear.
In fact there was just a single line missing who was meant to keep the
suggestion frame version up to date with the latest workspace state.


![lens_suggestion_bug_fix](https://github.com/user-attachments/assets/ff59e4d6-cfac-4801-857a-a79b1c98762d)

I've tried to add a unit test but it was so hard to actually follow the
code underneath, so I settled to add a FTR, which led to few more
extras:

* subtype selector now has test ids
* Lens page now has few more utility helpers for layer cloning and sub
type selector

### Checklist

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

delanni committed:
skip failing suite (#205141)

paul-tavares committed:
[Security Solution][Endpoint] Updates to enable background tasks and registered Fleet server extension in support of spaces (#221055)

## Summary

Change are in support of Endpoint space awareness and most of them are
to Fleet services which are used from within the Security Solution
plugin.

### Fleet Changes

- Several modules were updated with `.debug()` statements in order to
better debug issues and understand code flow.
- Some modules where updated to ensure that calls to SO Client methods
handle errors better (using a new Error utility named
`catchAndWrapError`)
- `PackagePolicy` and `AgentPolicy` services `.get()` methods where
updated to internally call `getById()` in order to support retrieving
policies with an un-scoped SO clients (allows for use of `namespaces:
[]` when using SoClient.bulkGet()`)

### Security Solution Changes:

- Manifest Manager was updated to
    - ensure it retrieves policies across all spaces 
- Ensure a properly scoped SO client is used when making updates -
including ensuring that when calling bulk update type of methods, the
updates are first grouped by space ID and then each group is called with
an SO client scoped to that space.
- Fleet server extension point callbacks were updated to ensure an
un-scoped SO client is used when they are invoked, so that policies can
be found in any space
- Policy Watcher and Policy Telemetry Watcher (background tasks) were
updated to ensure they retrieve policies across all spaces and if
needing ot update policies, that is done using a properly scoped
SoCLient.
- Several `.debug()` statements were sprinkled through out in order to
better understand code flow and failures that are logged to the kibana
log

jeramysoucy committed:
Upgrade prismjs 1.29.0 → 1.30.0 (#222019)

## Summary

Upgrades the `prismjs` dependency to version 1.30.0 where possible.

elena-shostak committed:
[CodeQL] Switched to a larger runner (#222060)

## Summary

Switched to a larger runner.

drewdaemon committed:
[ES|QL] suggest pipe after single column in EVAL (#221851)

## Summary

Previously, we weren't considering an assignment with a single field
name to be a "complete" expression in `EVAL`. This changes that because
it seems like people are using `EVAL` for this, even though we also have
a `RENAME` command.



https://github.com/user-attachments/assets/eab58527-ebfa-45bc-abb1-63b996ebbeac

kdelemme committed:
feat(streams): requires all spaces on kibana privilege (#221982)

## Summary

Resolves https://github.com/elastic/streams-program/issues/296

Streams kibana privilege can only be setup on All Spaces

| Other space | All Spaces |
|--------|--------|
| <img width="780" alt="image"
src="https://github.com/user-attachments/assets/83eb6a09-9345-48a1-aebc-92a5ab8c3e99"
/> | <img width="781" alt="image"
src="https://github.com/user-attachments/assets/09cb1fae-38fc-4e96-bc39-bcb98954cb01"
/> |

cesco-f committed:
[Synthetics] Skipping synthetics alerting tests on cloud (#222058)

mariairiartef committed:
[TSVB/SCSS] Migrate all remaining `.scss` styles to Emotion CSS-in-JS (#221484)

## Summary

Part of https://github.com/elastic/kibana/issues/220095

Migrates the following files to Emotion CSS-in-JS:
- `_gauge.scss`
- `_metric.scss`
- `_top_n.scss` 
- `_variables.scss`
- `_error.scss`
- `_vis_editor_visualization.scss`
- `_panel_config.scss`

### Considerations

For details on considerations, read
[this](https://github.com/elastic/kibana/issues/220095#issuecomment-2911737679).

### Checklist

- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

Omolola-Akinleye committed:
[Cloud Secutity Posture] Add  cloud connector support for asset discovery (#221689)

## Summary

Add Cloud Connector support for Agentless Asset Discovery Integration.
In this PR, we add the following:

- Feature flag in Security Solution Plugin to enable Cloud Connector
feature in Asset Discover Integration
- Add Cloud Connector as the default supported option in Agentless AWS
Form similar CSPM

<img width="1146" alt="Screenshot 2025-05-27 at 3 56 25 PM"
src="https://github.com/user-attachments/assets/b6b92136-ff24-4eed-ad95-3e1345dd70b8"
/>

<img width="1687" alt="Screenshot 2025-05-27 at 3 56 40 PM"
src="https://github.com/user-attachments/assets/434d4606-f3ea-4db0-a17a-8ba02fac314a"
/>

<img width="1665" alt="image"
src="https://github.com/user-attachments/assets/b7ce097c-e800-4a2f-9035-9382a3a6e07b"
/>

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

Supplementing committed:
[Fleet] Fix malformed MIGRATE action (#222081)

## Summary

Fix for MIGRATE actions with malformed JSON. Moves the
`enrollment_token`, `target_uri`, and `settings` into the data field of
the new action before creation so that fleet server can consume it
correctly.


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

### Identify risks

N/A

---------

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

Heenawter committed:
[Dashboard] Add collapsible sections (#220877)

Closes https://github.com/elastic/kibana/issues/1547
Closes https://github.com/elastic/kibana/issues/190342
Closes https://github.com/elastic/kibana/issues/197716

## Summary

This PR adds the ability for collapsible sections to be created and
managed on Dashboards.




https://github.com/user-attachments/assets/c5c046d0-58f1-45e1-88b3-33421f3ec002

> [!NOTE]
> Most of the work for developing collapsible sections occurred in PRs
contained to the `kbn-grid-layout` package (see [this meta
issue](https://github.com/elastic/kibana/issues/190342) to track this
work) - this PR simply makes them available on Dashboards by adding them
as a widget that can be added through the "Add panel" menu. As a result
of this, most work is contained in the Dashboard plugin - changes made
to the `kbn-grid-layout` package only include adding IDs for additional
tests that were added for the Dashboard integration.

### Technical Details

#### Content Management Schema

The content management schema allows for panels and sections to be mixed
within the single `panels` key for a dashboard **without** worrying
about section IDs; for example:

```
{
  "panels": [
    {
       // this is a simplified panel
       "gridData": {
         "x": 0,
         "y": 0,
         "w": 12,
         "h": 8,
       },
      "panelConfig": { ... },
    },
    {
       // this is a section
       "gridData": {
         "y": 9,
       },
      "collapsed": false,
      "title": "Section title",
      "panels": [
          {
          // this is a simplified panel
          "gridData": {
           "x": 0,
           "y": 0,
           "w": 24,
           "h": 16,
          },
          "panelConfig": { ... },
        },
      ],
   },
  ]
} 
```

#### Saved Object Schema

The dashboard saved object schema, on the other hand, separates out
sections and panels under different keys - this is because, while we are
stuck with panels being stored as `panelJSON`, I didn't want to add much
to this. So, under grid data for each panel, they have an optional
`sectionId` which then links to a section in the `sections` array in the
saved object:

```
{
  "panelsJSON": "<...> \"gridData\":{\"i\":\"panelId\",\"y\":0,\"x\":0,\"w\":12,\"h\":8,\"sectionId\":\"someSectionId\"} <...>"
  "sections": [
     {
       "collapsed": false,
       "title": "Section title",
       "gridData": {
          "i": "someSectionId",
          "y": 8.
        }
     }
  ],
}
```

This allows sections to be serialized **without** being stringified.
This storage also matches how we store this data in runtime using
`layout`.

### Checklist

- [x] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)


## Release note

Adds collapsible sections to Dashboard, which allow panels to grouped
into sections that will not load their contents when their assigned
section is collapsed.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

doakalexi committed:
[ResponseOps] Fixes failing test: Jest Integration Tests.x-pack/platform/plugins/shared/task_manager/server/integration_tests - unrecognized task types should be no workload aggregator errors when there are removed task types (#221854)

Resolves https://github.com/elastic/kibana/issues/208459

## Summary

Unskips flaky test that was failing with the error below. This PR
updates the test to wait until the task is finished running before
checking that the status is marked as `unrecognized`.
```
Error: expect(received).toBe(expected) // Object.is equality

Expected: "unrecognized"
Received: "idle"
    at toBe (/opt/buildkite-agent/
```

stratoula committed:
[ES|QL] [Lens] Renders the datagrid only when accordion is open (#222077)

tsullivan committed:
[Solution Nav] Functionality fixes (#220580)

## Summary

Epic: https://github.com/elastic/kibana-team/issues/1439

Closes https://github.com/elastic/kibana/issues/210221
Closes https://github.com/elastic/kibana/issues/219679

Changes
1. a11y fixes
   - `aria-label` to nav item links
   - `aria-expanded` for panel opener items
   - `aria-labelledby` for list elements
1. Add telemetry event tracking links in secondary panel
1. Fix cmd+click for recents and links in secondary panel
1. (maintenance) Rename `DefaultContent` => `Panel`
1. (maintenance) Move `navigation_section_ui` Emotion styles to separate
file

Ikuni17 committed:
skip failing test suite (#222124)

ersin-erdal committed:
Use msearch to fetch the alerts for maintenance windows with scoped query (#221702)

Currently we use a search with a `top_hits` aggregation to get the
alerts filtered with the scoped queries of the active maintenance
windows. But since `top_hits` aggregation has a fixed limit of 100 hits
we cannot use our max alerts limit to get all the alerts an execution
can generate.

This PR replaces `search` with `msearch` and removes the aggregation
from the search query so we can use the maxAlerts limit for the response
size.

jasonrhodes committed:
Correctly nests apm > synthetics serverless nav (#222115)

Fixes #222112

## Summary

I used the Machine Learning section as a guide because it has a similar
mix of "items" and "groups" but is presented correctly in the UI, so by
making the "apm" section look the same, it is fixed.

<img width="496" alt="Screenshot 2025-05-30 at 4 08 24 PM"
src="https://github.com/user-attachments/assets/e6a9d659-df58-4ec5-bf2d-2584fa2fe0d2"
/>

## Context

Machine Learning nav:

<img width="515" alt="Screenshot 2025-05-30 at 3 48 17 PM"
src="https://github.com/user-attachments/assets/25b7c1e4-8a99-43a4-be75-3299bd5412bd"
/>

| Top-level Group | Items | Sub-groups |
| :-- | :-- | :-- |
| Machine Learning | Overview, Data Visualizer | Anomaly Detection, Data
Frame Analytics, AIOps Labs |
| Applications | Service Inventory, Traces, Dependencies, Settings |
Synthetics |

I noticed the structure of these two sections was slightly different.
The machine learning one looks like this (i18n functions removed for
readability):

```ts
{
  id: 'machine_learning-landing',
  renderAs: 'panelOpener',
  title: 'Machine Learning',
  children: [
    {
      children: [
        { link: 'ml:overview' },
        { link: 'ml:dataVisualizer' },
      ],
    },
    {
      id: 'category-anomaly_detection',
      title: 'Anomaly detection',
      breadcrumbStatus: 'hidden',
      children: [
        { link: 'ml:anomalyExplorer' },
        { link: 'ml:singleMetricViewer' },
      ],
    },
  ]
}
```

whereas "Applications" included the synthetics sub-section further
nested in, as a sibling to the other items:

```ts
{
  id: 'apm',
  title: 'Applications',
  renderAs: 'panelOpener',
  children: [
    {
      children: [
        {
          link: 'apm:services',
          title: 'Service Inventory',
        },
        { link: 'apm:traces' },
        { link: 'apm:dependencies' },
        { link: 'apm:settings' },
        {
          id: 'synthetics',
          title: 'Synthetics',
          children: [
            {
              title: 'Overview',
              id: 'synthetics-overview',
              link: 'synthetics:overview',
              breadcrumbStatus: 'hidden',
            },
```

When I moved the Synthetics sub-section out of the children > children
section and into children, the nav rendered correctly.

I'm not sure if this is expected behavior, [there is some logic to
confirm](https://github.com/elastic/kibana/blob/0fcc9a39ba9d32b405172677a7df3c51c53c538a/src/platform/packages/shared/shared-ux/chrome/navigation/src/ui/components/panel/default_content.tsx#L75)
sections are "all items" OR "all groups" but ours seemed to be a mix,
but didn't fail.

kibanamachine committed:
[api-docs] 2025-05-31 Daily api_docs build (#222134)

Generated by
https://buildkite.com/elastic/kibana-api-docs-daily/builds/1089

shahzad31 committed:
[Synthetics] Remove run soon for sync private location task !! (#222062)

## Summary

Remove run soon for sync private location task !!

It's not needed since it's a scheduled task running every 10 minutes .

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

kibanamachine committed:
[api-docs] 2025-06-01 Daily api_docs build (#222138)

Generated by
https://buildkite.com/elastic/kibana-api-docs-daily/builds/1090

kertal committed:
[Inspect] Fix indexpattern parsing leading to incomplete index pattern values being displayed (#221084)

Fixes the parsing of index patterns in the Inspect feature. Previously, certain index pattern strings were not
being parsed and displayed correctly.

kibanamachine committed:
[ES|QL] Update function metadata (#222143)

This PR updates the function definitions and inline docs based on the
latest metadata from Elasticsearch.

Co-authored-by: Stratoula Kalafateli <efstratia.kalafateli@elastic.co>

kibanamachine committed:
[api-docs] 2025-06-02 Daily api_docs build (#222144)

Generated by
https://buildkite.com/elastic/kibana-api-docs-daily/builds/1091

drewdaemon committed:
[ES|QL] support `SAMPLE` command (#221446)

## Summary

Closes https://github.com/elastic/kibana/issues/217977

<img width="820" alt="Screenshot 2025-05-28 at 11 38 57 AM"
src="https://github.com/user-attachments/assets/fcee1c7c-3c90-4d1d-9e10-0453a8c3307f"
/>

### Open questions
- Should we be suggesting anything for `seed`?
- ~Do the suggestions for sampling rate make sense?~ (approved)

---------

Co-authored-by: Stratoula Kalafateli <efstratia.kalafateli@elastic.co>

dgieselaar committed:
Refine heuristic

dgieselaar committed:
Refine heuristic

elena-shostak committed:
Revert "[CodeQL] Switched to a larger runner" (#222148)

Reverts elastic/kibana#222060

agusruidiazgd committed:
[Security Solution][Network Page] a11y: Radio button options instantly closes when navigating through them (#220338)

## Summary

Closes: https://github.com/elastic/kibana/issues/205589


https://github.com/user-attachments/assets/762a3a01-90f8-4741-8ddd-b5b9a7fe65cc

Changes:

**Removed closing logic from `onChange`**
Deleted the `closePopover()` invocation in the radio-group’s `onChange`
Now selecting a radio option leaves the popover open as expected.

**Preserved Escape behavior**
The popover still closes when pressing Escape, per standard overlay
patterns.

**Accessibility alignment**
Followed the ARIA radio-group
[pattern](https://www.w3.org/WAI/ARIA/apg/patterns/radio/):
`Space/Enter` should select without closing, arrow keys move focus &
selection.
This matches both screen-reader usage and standard keyboard navigation.


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

---------

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

mbondyra committed:
[Maps] migrate vector to emotion (#217531)

## Summary
For core changes, exported 'EmotionStyles' type since it allows us to do
early ts check.

Part of https://github.com/elastic/kibana/issues/207852

darnautov committed:
[ML] Trained Models: Fix inference endpoints assignment  (#222076)

## Summary

Fixes the inference endpoint assignment to the trained model object. The
`elser` service was replaced with `elasticsearch`, so the check for
assigning endpoints wasn’t working.

It fixes the check for the Stop deployment action on the Trained Models
page, preventing the user from stopping an inference endpoint.


<img width="771" alt="image"
src="https://github.com/user-attachments/assets/1c76a716-971f-410a-a82a-141dc3d9b9fc"
/>


### Checklist

- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

---------

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

kibanamachine committed:
[ES|QL] Update grammars (#222142)

This PR updates the ES|QL grammars (lexer and parser) to match the
latest version in Elasticsearch.

---------

Co-authored-by: Stratoula Kalafateli <efstratia.kalafateli@elastic.co>

juliaElastic committed:
[Fleet] show loading until package title loads (#221784)

## Summary

Closes https://github.com/elastic/kibana/issues/221760

Replace package name with loading spinner until package info is loaded



https://github.com/user-attachments/assets/bb5612c9-95e9-4a77-9e12-f03f117583aa

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

rmyz committed:
[ObsUX][APM][Infra] Use profiling plugin configuration instead of feature flags (#219904)

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

jennypavlova committed:
[APM] Fix error count waterfall navigation reload issue (#221664)

Closes #82154 

## Summary

This PR fixes the error count navigation reload issue in the APM
waterfall. To fix that, it combines the prevent default action needed
because of the flyout opening and the navigation using the
`navigateToUrl` action to create a SPA-like experience:


https://github.com/user-attachments/assets/43faf3b9-2038-40d0-89c9-62c37087386d

Testing ⬆️ 
- The navigation to the errors tab should not cause a full page refresh
- The flyout should open without issues on click 

+ Now the errors link supports keyboard navigation (fixed that part
here, it was skipped before)


https://github.com/user-attachments/assets/82c2d0b7-caef-4c24-8af9-d6b1d7f75eef

yngrdyn committed:
[Dataset quality] unskipping  summary tests (#222172)

Closes https://github.com/elastic/kibana/issues/211516

yngrdyn committed:
[Dataset quality] unskipping stats api tests (#222178)

Closes https://github.com/elastic/kibana/issues/211517 and
https://github.com/elastic/kibana/issues/213290

afharo committed:
chore(): unify solution id type definition (#222008)

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

rmyz committed:
[APM] Skip failing test (#222185)

dgieselaar committed:
Merge branch 'streams-agent' of github.com:dgieselaar/kibana into streams-agent
